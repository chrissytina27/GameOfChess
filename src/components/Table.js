import React from "react";
import PropTypes from "prop-types";
import { util } from "../helpers/util";

const Table = ({ players }) => {
  const renderTable = () => {
    const whiteMoves = players[1].moves,
      blackMoves = players[0].moves;
    const table = whiteMoves.map((move, index) => {
      const whiteMove = whiteMoves[index] && whiteMoves[index].to;
      const blackMove = blackMoves[index] && blackMoves[index].to;
      return (
        <tr key={index}>
          <td>{whiteMove && util.squareName(whiteMove)}</td>
          <td>{blackMove && util.squareName(blackMove)}</td>
        </tr>
      );
    });
    return <tbody>{table}</tbody>;
  };

  return (
    <table>
      <thead>
        <tr>
          <th>White</th>
          <th>Black</th>
        </tr>
      </thead>
      {renderTable()}
    </table>
  );
};
Table.propTypes = {
  players: PropTypes.array
};
export default Table;
