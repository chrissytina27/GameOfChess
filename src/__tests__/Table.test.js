import Table from "../components/Table";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

describe("Table", () => {
  const testProps = [
    {
      graveyard: [],
      graveyardPick: false,
      name: "player1",
      moves: ["d5", "c3", "e333"]
    },
    {
      graveyard: [],
      graveyardPick: false,
      name: "player2",
      moves: ["d5", "c3", "e333"]
    }
  ];
  it("renders correctly", () => {
    const tree = renderer.create(<Table players={testProps} />).toJSON();
    expect(tree.children[1].children.length).toBe(testProps[1].moves.length);
    expect(tree).toMatchSnapshot();
  });
});
