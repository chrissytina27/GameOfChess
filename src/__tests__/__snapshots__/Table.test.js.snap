// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        White
      </th>
      <th>
        Black
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td />
      <td />
    </tr>
    <tr>
      <td />
      <td />
    </tr>
    <tr>
      <td />
      <td />
    </tr>
  </tbody>
</table>
`;
